### DISCUSSION TITLE
Expand lookups in datamodel

### DISCUSSION THREAD

**Discussion by nebuchadnezzar-network** [netbox-community/netbox#10532] [General]
Hi,
This might be a noob question. 

I'm using the API for the first time and so far it works as expected. However I try to get the DNS name of a virtual machine (Interfaces -> IP address -> DNS name) 

Is this possible to expand/children/lookup foreign keys using the API in one call? 

I have read the docs but it does not talk to much the topic. Any hints would be helpful. 

Thanks!

**Comment by candlerb** [comment]
> Is this possible to expand/children/lookup foreign keys using the API in one call?

Not in the REST API, but it might be possible with [GraphQL](https://docs.netbox.dev/en/stable/integrations/graphql-api/).

Note that it would be good practice to set a "primary_ip" on the virtual machine, so in this example it would be only two lookups: one for the VM (which returns the primary IP), and one for the primary IP (which gives you the DNS name).

**Reply by nebuchadnezzar-network** [reply]
Thanks, 
Haven't looked at GraphQL. 

Yes, all VMs have primary both IPv4 and IPv6 but was not sure if the primary IP(s) where listed in the API. 
It would, at least from what CMDB tools I've used in the past useful to have the FQDN(s) on all Physical and Virtual CIs. 

The use case here is to enable monitoring by installing agents and configure SNMP and for that I just need to know the FQDN of certain interfaces.
